api_server.py

```
@@ -27,11 +27,23 @@
+        if "db" not in g:
+            g.db = sqlite3.connect(DB_PATH, detect_types=sqlite3.PARSE_DECLTYPES)
+            g.db.row_factory = sqlite3.Row
+        return g.db
+    # Fallback for standalone scripts/tests outside request context
     conn = sqlite3.connect(DB_PATH, detect_types=sqlite3.PARSE_DECLTYPES)
     conn.row_factory = sqlite3.Row
     return conn
 
 
+@app.teardown_appcontext
+def close_db(error):
```

Ruff (0.15.0)

[warning] Unused function argument: `error`

(ARG001)